property/index.js

```diff
@@ -115,54 +119,133 @@ module.exports = yeoman.generators.Base.extend({
       {
          name: 'defaultValue',
          message: 'Default value[leave blank for none]:',
-         default: null
+         default: null,
+         when: function(answers) {
+          return answers.type !== null && 
+            answers.type !== 'buffer' &&
+            answers.type !== 'any' &&
+            typeChoices.indexOf(answers.type) !== -1;
+        }
       }
     ];
+
     this.prompt(prompts, function(answers) {
+      debug('answers: %j', answers);
       this.name = answers.name || this.name;
       if (answers.type === 'array') {
         var itemType =  answers.customItemType || answers.itemType;
         this.type = itemType ? [itemType] : 'array';
       } else {
         this.type = answers.customType || answers.type;
       }
-      this.required = answers.required;
-      this.defaultValue = answers.defaultValue;
+
+      this.propDefinition = {
+        name: this.name,
+        type: this.type,
+      };
+      if (answers.required) {
+        this.propDefinition = Boolean(answers.required);
```

Comment:
I think I have already asked about this, but why cannot we simply set `this.propertyDefinition.defaultValue = answers.required`, no conversion needed?

Also notice you are overwriting whole `this.propDefinition` object here, that's probably not what you want?